### PR TITLE
Update to wgpu 0.20

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This is a draft while I track a couple of things that need to happen (like wgpu-profiler update).

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
This is blocked on https://github.com/Wumpf/wgpu-profiler/pull/67 and a new release of that crate.

**Comment by waywardmonkeys** [comment]
I've checked this on macOS. Would be great if someone could check it on Linux and/or Windows.

**Review by DasLixou — APPROVED** [review]
Code looks ok, on Windows/AMD I have big gaps but that could be from AMD drivers.
https://xi.zulipchat.com/user_uploads/8784/pYDOohkUsiQOCnx6QZ7JW6Jd/20240513-1226-26.9507645.mp4

File: src/wgpu_engine.rs [hunk 1]
```diff
@@ -644,6 +644,7 @@ impl WgpuEngine {
             layout: Some(&compute_pipeline_layout),
             module: &shader_module,
             entry_point: "main",
+            compilation_options: PipelineCompilationOptions::default(),
```
Comment (DJMcNab) [inline src/wgpu_engine.rs:647]: We will want to disable zero initialisation here, but that can be a followup

**Review by DJMcNab — APPROVED** [review]
Tested on Vulkan (Linux, Mesa Intel) and Vulkan (Android - Pixel 6)

Both work fine, thanks!

**Comment by waywardmonkeys** [comment]
@DJMcNab You're okay with this landing now? You said something about "after office hours".

**Comment by DJMcNab** [comment]
Yes, this should land now

"After office hours" is for the release of Vello 0.2.0

**Pull request merged**